### DISCUSSION TITLE
Feature request: Add input type `file` to upload files to workflow runs.

### DISCUSSION THREAD

**Discussion by codiophile** [community/community#49612] [Actions]
### Select Topic Area

Product Feedback

### Body

I have a Jenkins pipeline that allows you to upload a file, that processes the uploaded file and creates an artifact out of the processed file. So far, I haven't found a way to migrate this pipeline to GitHub Actions, as it doesn't seem like there is a way to upload files to workflows. Obviously, I could upload the file to Nexus or similar and then provide the URL as input to GitHub Actions, but that would add an extra step and for as long as the Jenkins job is still available, I would always use that.

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_dispatchinputsinput_idtype

**I think the cleanest way to add this functionality would be to add a `file` input type to workflow dispatch.** 

I suppose you could also provide an action that lets you upload a file, but then you would have to trigger the workflow and wait for a runner to pick it up and reach that action, which is certainly a worse experience. I'm not sure if it is possible to write actions that integrate with the GitHub UI. I'm assuming it isn't, but if it is, the community could probably write an action that does this. So if anything like that already exists, please let me know.

**Comment by ooaklee** [comment]
TLDR; 

Hey @codiophile, and for anyone else who might be interested, I've created [an action - **Interactive Inputs**](https://github.com/boasiHQ/interactive-inputs) that gives you Runtime inputs, and it also supports `file` and `multifile` input types that you can leverage in your workflows.

Please check it out and let me know your thoughts, feedback and future feature requests; you can [**find it on the marketplace**](https://github.com/marketplace/actions/interactive-inputs).

======

See in action below (without sending  portal Url to Slack/Discord) -  ![InteractiveInputs_LiveFileInputTypeSupport](https://github.com/user-attachments/assets/d57ae28e-3d0e-476b-8146-480f17bcf271)

## Example workflow.yaml using action with `multifile` input field

Below is an example of how you can leverage this action in your workflow file, you will neeed to replace the `NGROK_AUTHTOKEN` secret with your own, and if you wish to send notifications to Slack/ Discord, you will need to replace the `SLACK_TOKEN` and `DISCORD_WEBHOOK` secrets with your own as well as enabling the `notifier-slack-enabled` and `notifier-discord-enabled` inputs.

```yaml
name: "Example Interactive Inputs Workflow"

on:
  workflow_dispatch:

jobs:
  interactive-inputs:
    timeout-minutes: 3
    runs-on: ubuntu-latest
    permissions:
      contents: write
    steps:
      - name: Example Interactive Inputs Step
        id: interactive-inputs
        uses: boasihq/interactive-inputs@v2
        with:
          ngrok-authtoken: ${{ secrets.NGROK_AUTHTOKEN }}
          notifier-slack-enabled: "false"
          notifier-slack-channel: "#notificaitons"
          notifier-slack-token: ${{ secrets.SLACK_TOKEN }}
          notifier-discord-enabled: "false"
          notifier-discord-webhook: ${{ secrets.DISCORD_WEBHOOK }}
          timeout: 160
          title: "Did someone ask for file input type support?"
          interactive: |
            fields:
              - label: requested-files
                properties:
                  display: Upload desired files
                  type: multifile
                  required: true
                  description: Upload desired files that are to be uploaded to the runner for processing

      - name: List the uploaded files in the directory
        shell: bash
        run: |
          echo "Display uploaded files"
          echo -e "\n==============================\n"
          ls -la ${{ steps.interactive-inputs.outputs.requested-files }} # Use the label of the multifile/file field as the key to get the uploaded file directory
          echo -e "\n==============================\n"

```
